### PR TITLE
Corrected authorization behavior for abort_job and added test cases

### PR DESCRIPTION
**Pull request by nvidianz**

### CODE REVIEW

File: nvflare/private/fed/server/job_cmds.py [hunk 1]
```diff
@@ -79,9 +79,19 @@ def authorize_job(self, conn: Connection, args: List[str]):
 
         job_id = args[1].lower()
         conn.set_prop(self.RUN_NUMBER, job_id)
-        args.append("server")  # for now, checking permissions against server
+        engine = conn.app_ctx
+        job_def_manager = engine.job_def_manager
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/server/job_cmds.py:83]: note that if this engine does not have `job_def_manager`
this L83 will raise `AttributeError`, we don't get to L84

File: nvflare/private/fed/server/job_cmds.py [hunk 2]
```diff
@@ -153,8 +163,9 @@ def delete_job(self, conn: Connection, args: List[str]):
 
     def abort_job(self, conn: Connection, args: List[str]):
         engine = conn.app_ctx
-        if not isinstance(engine, ServerEngine):
-            raise TypeError("engine must be ServerEngine but got {}".format(type(engine)))
+        job_runner = engine.job_runner
+        if not job_runner:
+            raise TypeError("engine missing job_runner")
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/server/job_cmds.py:168]: note that if this engine does not have `job_runner`
this L166 will raise AttributeError, we don't get to L167

File: tests/unit_test/fuel/hci/client/admin_auth_test.py [hunk 3]
```diff
@@ -17,7 +17,7 @@
 import os
 import time
 import uuid
-from typing import List
+from typing import List, Optional
 from unittest.mock import MagicMock, Mock
 
 from nvflare.apis.fl_constant import WorkspaceConstants
```
Comment (YuanTingHsieh) [inline tests/unit_test/fuel/hci/client/admin_auth_test.py:23]: can you run `isort` to fix the import order in this file? thanks

File: nvflare/private/fed/server/job_cmds.py (hunk 1 above)
Comment (nvidianz) [reply nvflare/private/fed/server/job_cmds.py:83]: AttributeError is just as meaningful so I am going to remove this check.

File: nvflare/private/fed/server/job_cmds.py (hunk 2 above)
Comment (nvidianz) [reply nvflare/private/fed/server/job_cmds.py:168]: Will remove this check.

File: tests/unit_test/fuel/hci/client/admin_auth_test.py (hunk 3 above)
Comment (nvidianz) [reply tests/unit_test/fuel/hci/client/admin_auth_test.py:23]: Fixed

**Review by chesterxgchen — APPROVED** [review]

**Pull request merged**